tests/testthat/test-process_request.R

```diff
@@ -0,0 +1,86 @@
+test_that("process_request properly errors on wrong content type", {
+
+  ## create fake res with the wrong content type
+  actual <- "test"
+  res <- list(headers = list("content-type" = actual))
+  expect_error(process_request(res, internet = FALSE),
+               sprintf(paste0("Expected content-type:\n%s",
+                              "\n",
+                              "Actual content-type:\n%s"
+               ),
+               "application/json; charset=UTF-8",
+               actual)
+  )
+
+  ## let's try if I feed it an expected type
+  actual = "test"
+  expected = "something else"
+  res <- list(headers = list("content-type" = actual))
+  expect_error(process_request(res, expected = expected, internet = FALSE),
+               sprintf(paste0("Expected content-type:\n%s",
+                              "\n",
+                              "Actual content-type:\n%s"
+               ),
+               expected,
+               actual)
+  )
+
+  ## let's try it if it's okay
+  actual = "application/json; charset=UTF-8"
+  res <- list(headers = list("content-type" = actual))
+
+  ## should return NULL
+  expect_equal(process_request(res, internet = FALSE), NULL)
+
+  ## also if we set a new expectation
+  actual = "test"
+  expected = "test"
+  res <- list(headers = list("content-type" = actual))
+
+  ## should return NULL
+  expect_equal(process_request(res,
+                               expected = expected,
+                               internet = FALSE),
+               NULL)
+
+})
+
+## testing on the API
+
+test_that("process_request properly errors on wrong content type", {
+
+  skip_on_appveyor()
+  skip_on_travis()
+
+  ## something we know won't be correct
+  res <- httr::GET("http://livefreeordichotomize.com")
+
+  expected = "application/json; charset=UTF-8"
+  actual = res$headers$`content-type`
+  ## let's make sure our test is actually good, this should be true
+  expect_true(actual != expected)
+
+  expect_error(process_request(res),
+               sprintf(paste0("Expected content-type:\n%s",
+                              "\n",
+                              "Actual content-type:\n%s"
+               ),
+               expected,
+               actual)
+  )
+
+  x <- drive_list()[1,] # grab most recent thing on drive
```

Comment:
I was trying to prove that it works with a generic call from our api, but maybe that is silly 😬, I can take out that last part